include/image.h

```diff
@@ -55,6 +60,282 @@
 #endif
 
 
+
+
+#if defined(ARCH_ARM) && defined(WOLFBOOT_ARMORED)
+
+struct wolfBoot_image {
+    uint8_t *hdr;
+    uint8_t *trailer;
+    uint8_t *sha_hash;
+    uint8_t *fw_base;
+    uint32_t fw_size;
+    uint32_t part;
+    uint32_t hdr_ok;
+    uint32_t canary_FEED4567;
+    uint32_t signature_ok;
+    uint32_t canary_FEED6789;
+    uint32_t not_signature_ok;
+    uint32_t canary_FEED89AB;
+    uint32_t sha_ok;
+};
+
+static void __attribute__((noinline)) wolfBoot_image_confirm_signature_ok(struct wolfBoot_image *img)
+{
+    img->canary_FEED4567 = 0xFEED4567;
+    img->signature_ok = 1;
+    img->canary_FEED6789 = 0xFEED6789;
+    img->not_signature_ok = ~(1);
+    img->canary_FEED89AB = 0xFEED89AB;
+}
+
+#define PART_SANITY_CHECK(p) \
+    asm volatile("mov r2, #0"); \
```

Comment:
I recommend using `__asm__ __volatile__`. Also you'll need macros for:

```
#ifdef __IAR_SYSTEMS_ICC__
    #define __asm__        asm
    #define __volatile__   volatile
#endif
#ifdef __KEIL__
    #define __asm__        __asm
    #define __volatile__   volatile
#endif
```

Reply:
This is no longer necessary as the armored code is arm-gcc only